app/models/user.rb

```diff
@@ -207,6 +207,11 @@ def index_id
     "users-#{id}"
   end
 
+  def set_remember_fields
+    self.remember_token ||= self.class.remember_token if respond_to?(:remember_token)
+    self.remember_created_at ||= Time.now.utc
```

Comment:
`Time.current` is fine, the server runs on UTC

Reply:
@rhymes What about servers that are not set in UTC? This is true for DEV but I thought the goal of this project going open source was so that other people might make their own community.

Reply:
@Glennmen that's why you should use time zone aware methods like `Time.current` in the first place. Because in the event that the server is not running on UTC all time objects will be time zone aware and possibly will use the same timezone.

In this case `Time.now.utc` is not wrong per se, but as you said if you switch the timezone of the server then `remember_created_at` will always be now in UTC and I don't think that's the goal. The goal here, I believe, is to set `remember_created_at` to what "now" means on the server. 

My best practice that has worked for me in years is to accept any time zone in input, possibly do operations in UTC in memory and then convert to whatever timezone the user wants in output, maybe by using [Time.use_zone](https://www.varvet.com/blog/working-with-time-zones-in-ruby-on-rails/) on a per user basis.

It's a tricky topic 😅 

If you want to read more about it:

* [Working with time zones in Ruby on Rails](https://www.varvet.com/blog/working-with-time-zones-in-ruby-on-rails/)
* [It's About Time (Zones)](https://thoughtbot.com/blog/its-about-time-zones)
* [The Exhaustive Guide to Rails Time Zones](http://danilenko.org/2012/7/6/rails_timezones/)

Reply:
I've copied these 2 lines from `devise` to keep existing functionality, and in case devise code is used somewhere implicitly inside the app.

Reply:
Got it. Devise uses `Time.now.utc` because they are not tied to Rails/ActiveRecord and `Time.zone.now` or `Time.current` are only available in `ActiveSupport` extensions. That line is going to get flagged by Rubocop, after #2334 gets merged, that's why I suggested the change.

I wish there was *one* way to deal with timezones so we didn't have all of these permutations to do the same thing :D

Reply:
> @rhymes What about servers that are not set in UTC? This is true for DEV but I thought the goal of this project going open source was so that other people might make their own community.

I think we can cross this bridge when we get there. We should err on the side of doing things that get us towards this goal, but IMO we shouldn't let it be a blocker for the task at hand though.

We'll get a roadmap of some sort out soon. I think we'll probably get to true generalizability by the one year anniversary of open source, August 8th.

Reply:
Having a server running in UTC is also something that can be a requirement of running the source code. I would expect most servers to be UTC though.